### PR TITLE
IA-1116 Selectable docker

### PR DESCRIPTION
**Pull request by Qi77Qi**
Related PR https://github.com/DataBiosphere/terra-docker/pull/1

Things should work for existing image as well

Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/resources/reference.conf [hunk 1]
```diff
@@ -2,7 +2,6 @@
 # Google Cloud dataproc configuration
 dataproc {
   applicationName = "firecloud:leonardo"
-  dataprocDockerImage = "gcr.io/broad-dsde-prod/leonardo-jupyter:prod"
```
Comment (Qi77Qi) [inline src/main/resources/reference.conf:5]: this is removed because it's duplicated by leonardo.conf

File: src/main/resources/reference.conf [hunk 2]
```diff
@@ -11,7 +10,15 @@ dataproc {
   networkTag = "leonardo"
   projectVPCNetworkLabel = "vpc-network-name"
   projectVPCSubnetLabel = "vpc-subnetwork-name"
-  defaultScopes = ["https://www.googleapis.com/auth/userinfo.email", "https://www.googleapis.com/auth/userinfo.profile", "https://www.googleapis.com/auth/bigquery", "https://www.googleapis.com/auth/source.read_only"]
+  defaultScopes = [
```
Comment (Qi77Qi) [inline src/main/resources/reference.conf:13]: adding this to reference.conf and we should remove this from leonardo.conf

File: src/main/resources/jupyter/init-actions.sh [hunk 3]
```diff
@@ -187,7 +187,7 @@ if [[ "${ROLE}" == 'Master' ]]; then
     # authorize docker to interact with gcr.io.
     if grep -qF "gcr.io" <<< "${JUPYTER_DOCKER_IMAGE}${RSTUDIO_DOCKER_IMAGE}${PROXY_DOCKER_IMAGE}${WELDER_DOCKER_IMAGE}" ; then
       log 'Authorizing GCR...'
-      gcloud docker --authorize-only
+      gcloud auth configure-docker
```
Comment (rtitle) [inline src/main/resources/jupyter/init-actions.sh:190]: 👍

File: src/main/resources/jupyter/init-actions.sh [hunk 4]
```diff
@@ -326,6 +329,8 @@ if [[ "${ROLE}" == 'Master' ]]; then
 
       fi
 
+       retry 5 docker exec -u root ${JUPYTER_SERVER_NAME} chown -R jupyter-user:users ${JUPYTER_HOME}
```
Comment (rtitle) [inline src/main/resources/jupyter/init-actions.sh]: Is this needed? I thought we do this in the Dockerfile.

File: src/main/resources/reference.conf (hunk 1 above)
Comment (rtitle) [reply src/main/resources/reference.conf:5]: Yeah.. makes sense.

**Review by rtitle — APPROVED** [review]
:shipit: pending tests

**Pull request merged**